source/common/formatter/substitution_formatter.cc

```diff
@@ -54,5 +54,104 @@ const re2::RE2& SubstitutionFormatParser::commandWithArgsRegex() {
   // clang-format on
 }
 
+JsonFormatBuilder::FormatElements
+JsonFormatBuilder::fromStruct(const ProtobufWkt::Struct& struct_format) {
+  output_.clear();
+
+  formatValueToFormatElements(struct_format.fields());
+  output_.push_back(FormatElement{std::move(buffer_), false});
+  ASSERT(buffer_.empty(), "buffer_ should be consumed");
```

Comment:
std::move does not guarantee clearing ref. Just call clear().

Reply:
From https://en.cppreference.com/w/cpp/utility/move : all standard library objects that have been moved from are placed in a "valid but unspecified state", meaning the object's class invariants hold (so functions without preconditions, such as the assignment operator, can be safely used on the object after it was moved from):

Reply:
got it.